### ISSUE TITLE
lab02 files were deleted

### ISSUE THREAD

**Issue by jasminacosta** [Watts-College/paf-515-spr-2024#4]
@castower 

Hi! When I pushed all my files to github it deleted all my work from lab02?

Not sure how to fix this

**Comment by castower** [comment]
Hi Jasmin, 

I can see your files on the repo that appear to have been updated 33 minutes ago (https://github.com/Watts-College/project-paf-515-2024s-team-04/tree/main/labs/wk02). Are there changes in the files that were removed?

**Comment by jasminacosta** [comment]
Hi! 

Yes, the files are in Github but on my computer they are completely gone along with the progress I made to my project_data_steps file. 

I have no idea why. Right now I am redownloading the files from Github and I can move them back to the github folder on my computer, but I am not sure how that will mess with the whole system especially for the Repo.

**Comment by castower** [comment]
@jasminacosta In GitHub Desktop does it show that you're synched to the main branch?

**Comment by jasminacosta** [comment]
Yes, I believe so.

When I pushed I did it to the "jasmin" branch like in lecture 1, but not sure if that was the problem I did.

<img width="943" alt="Screenshot 2024-03-21 at 4 53 20 PM" src="https://github.com/Watts-College/paf-515-spr-2024/assets/136217967/ef56e6db-e0b0-4cb8-9701-9dd2e48b7048">

**Comment by castower** [comment]
@jasminacosta 

The snapshot here is showing your local computer is two commits behind main (see the arrow and number 2 next to pull origin). Since your files are on main you should be able to pull them down with the pull origin button.

It's also best to start fresh with a new branch for each lab just to ensure your branches are always up-to-date with main.

Let me know if you run into any other issues!

**Comment by jasminacosta** [comment]
I also submitted all the dowloaded files that way if I needed to download them again, I will be able to since it has all my work

**Comment by jasminacosta** [comment]
@castower In this case

1. pull the origin 
2. and create a new branch like "jasmin lab2" as an example? since I already have a branch named "jasmin"
3. when i pressed pull origin I came across the following message:

<img width="883" alt="Screenshot 2024-03-21 at 5 03 33 PM" src="https://github.com/Watts-College/paf-515-spr-2024/assets/136217967/a88404b4-a519-4f5f-a134-5b3261428dc4">

Not sure if that is fine to override to be able to get my files back to my local computer

**Comment by castower** [comment]
@jasminacosta 

I'm not sure if the .DS_STORE file would impact anything on your local machine, but you can close the message box and then ignore the .DS_Store file by adding it to your .gitignore since it won't impact anything for the project.

Example:
```
# Ignore .DS_STORE folder
.DS_STORE
```
That should remove those files from your git tracking and then you can do the pull origin.

Next, since you've already pushed lab02 you shouldn't need another branch, but yes for all after you can label the branch with your name and the lab number.

**Comment by jasminacosta** [comment]
@castower I placed the following code provided to the .gitignore file, but it is still showing on Github desktop with the same message appearing.

**Comment by castower** [comment]
@jasminacosta 
It must be finicky because it's already on the repo, I'll remove the file. 

Did your current .DS_STORE files disappear from your changed files list?

**Comment by jasminacosta** [comment]
@castower I dont see the .DS_STORE files now that I am looking at my folder

**Comment by castower** [comment]
@jasminacosta  Okay, I removed it from main. Try now and see if you're able to complete the pull

**Comment by jasminacosta** [comment]
This is the message I am receiving now, but it seems like it would overwrite the file I have right now which does not have my updated work and it would overwrite my work that is in Github?

<img width="920" alt="Screenshot 2024-03-21 at 5 30 34 PM" src="https://github.com/Watts-College/paf-515-spr-2024/assets/136217967/43819a42-4551-4f15-915e-6b02cef9d976">

<img width="670" alt="Screenshot 2024-03-21 at 5 31 13 PM" src="https://github.com/Watts-College/paf-515-spr-2024/assets/136217967/205e5215-8410-46eb-a371-0012f2e85f4f">

**Comment by castower** [comment]
@jasminacosta Yes, it should overwrite. Just to be certain you don't lose anything, I would back up the files listed here to another folder and then complete the pull.

**Comment by jasminacosta** [comment]
When I tried to pull this is the step I am at now

<img width="924" alt="Screenshot 2024-03-21 at 5 42 21 PM" src="https://github.com/Watts-College/paf-515-spr-2024/assets/136217967/9226f328-9b6f-4fda-afc2-13b8946429ce">

**Comment by castower** [comment]
@jasminacosta, 
Okay, you can now do the following:

1) Discard your stashed changes since those were the wrong files for your lab 
2) Re-update your .gitignore to ignore the .DS_Store files
3) Commit this updated .gitignore

You should then be all up-to-date.

**Comment by jasminacosta** [comment]
@castower I see my files back.

Thank you so much for all your help!!

**Comment by castower** [comment]
Yay! No problem :)

**Comment by jasminacosta** [comment]
😄
